# Water Crisis Analysis in Maji Ndogo – Part 3: Data Validation, Auditing & Investigative Analysis

## Introduction

This notebook represents Part 3 of a four-part SQL analysis project focused on water accessibility and infrastructure in Maji Ndogo.

While previous sections explored and prepared the dataset, this phase focuses on **data validation, auditing, and investigative analysis**. The objective is to assess the integrity of the dataset by comparing internally collected data with an independent auditor’s report, identifying inconsistencies, and investigating potential sources of error.

This stage is critical in ensuring that insights derived from earlier analyses are reliable and can support informed decision-making.

---

## Objectives

The key objectives of this analysis are:

- Integrate the independent auditor’s dataset into the existing database  
- Validate the accuracy of recorded water quality scores  
- Identify discrepancies between surveyor and auditor assessments  
- Investigate patterns in incorrect records  
- Link discrepancies to responsible entities (employees)  
- Detect anomalies that may indicate systematic errors or potential misconduct  

---

## Context

An independent audit was conducted to verify the accuracy and integrity of the water services database. The audit findings revealed that while most records align with expected standards, a subset of records shows inconsistencies that require further investigation.

This notebook focuses on identifying and analysing these discrepancies to ensure data reliability and accountability.

---

## Structure of This Notebook

The analysis is structured as follows:

1. Understanding database relationships (ERD context)  
2. Integrating the auditor’s report  
3. Comparing surveyor and auditor scores  
4. Identifying inconsistent records  
5. Linking discrepancies to employees  
6. Quantifying error patterns  
7. Identifying high-risk or anomalous behaviour  
8. Extracting supporting evidence from qualitative data  

---

## Key Consideration

Data validation is a crucial step in any analytical workflow. Inaccurate or manipulated data can lead to misleading conclusions and poor decision-making. This analysis ensures that the dataset used in subsequent insights is both accurate and trustworthy.

In [1]:
import pandas as pd
from getpass import getpass

In [2]:
%load_ext sql

In [3]:
password = getpass('Enter DB password: ')

%sql mysql+pymysql://root:{password}@localhost:3309/md_water_services

Enter DB password:  ········


'Connected: root@md_water_services'

### Integrating the Auditor’s Report

An external audit dataset is introduced to validate the reliability of the existing data. This dataset contains independently assessed water quality scores and supporting observations. The auditor’s report is imported into the database and structured to align with existing tables, enabling direct comparison with internally collected data.

In [4]:
%%sql

DROP TABLE IF EXISTS `auditor_report`;
CREATE TABLE `auditor_report` (
`location_id` VARCHAR(32),
`type_of_water_source` VARCHAR(64),
`true_water_source_score` int DEFAULT NULL,
`statements` VARCHAR(255)
);

 * mysql+pymysql://root:***@localhost:3309/md_water_services
0 rows affected.
0 rows affected.


[]

In [5]:
df = pd.read_csv('Auditor_report.csv', delimiter = ';') 

In [6]:
df.head()


,location_id,type_of_water_source,true_water_source_score,statements
0,SoRu34980,well,1,Residents admired the official's commitment to...
1,AkRu08112,well,3,Villagers spoke highly of the official's dedic...
2,AkLu02044,river,0,Villagers were touched by the official's inter...
3,AkHa00421,well,3,"Villagers were moved by the official's visit, ..."
4,SoRu35221,river,0,"A photographer's lens captures the queue, thou..."


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1620 entries, 0 to 1619
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   location_id               1620 non-null   object
 1    type_of_water_source     1620 non-null   object
 2    true_water_source_score  1620 non-null   int64 
 3    statements               1620 non-null   object
dtypes: int64(1), object(3)
memory usage: 50.8+ KB


The dataset above contains 1,620 records representing locations that were revisited as part of the audit process. Each record includes the `location_id`, the type of water source at that location, and an independently assessed water quality score. In addition to quantitative measurements, the auditor collected qualitative statements from local residents, providing contextual insights into each site.

To address the first objective, it is necessary to compare the water quality scores recorded in the `water_quality` table with those provided in the `auditor_report`. While the auditor’s dataset is indexed by `location_id`, the `water_quality` table is linked via `record_id`. The `visits` table serves as the bridge between these two identifiers, enabling accurate linkage between the auditor’s assessments and the corresponding survey records.

In [8]:
df.columns.tolist()

['location_id',
 ' type_of_water_source',
 ' true_water_source_score',
 ' statements']

In [9]:
import pandas as pd
import pymysql

conn = pymysql.connect(
    host="localhost",
    port=3309,
    user="root",
    password=password,
    database="md_water_services"
)

cursor = conn.cursor()

sql = """
INSERT INTO auditor_report
(location_id, type_of_water_source, true_water_source_score, statements)
VALUES (%s, %s, %s, %s)
"""

cursor.executemany(sql, df.values.tolist())
conn.commit()

print(f"{cursor.rowcount} rows inserted.")

cursor.close()
conn.close()

1620 rows inserted.


In [10]:
%%sql 
SELECT *
FROM auditor_report
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
5 rows affected.


location_id,type_of_water_source,true_water_source_score,statements
SoRu34980,well,1,"Residents admired the official's commitment to enhancing urban life, praising their cooperative and inclusive approach."
AkRu08112,well,3,"Villagers spoke highly of the official's dedication and genuine interest in their lives, fostering a sense of belonging and appreciation."
AkLu02044,river,0,"Villagers were touched by the official's interactions, noting their humility, strong work ethic, and respectful attitude."
AkHa00421,well,3,"Villagers were moved by the official's visit, praising their hard work, humility, and the profound sense of connection they fostered."
SoRu35221,river,0,"A photographer's lens captures the queue, though his own struggle for water is a hidden part of the story."


In [11]:
%%sql

SHOW TABLES;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
10 rows affected.


Tables_in_md_water_services
Incorrect_records
auditor_report
data_dictionary
employee
global_water_access
location
visits
water_quality
water_source
well_pollution


In [12]:
%%sql

DESCRIBE auditor_report

 * mysql+pymysql://root:***@localhost:3309/md_water_services
4 rows affected.


Field,Type,Null,Key,Default,Extra
location_id,varchar(32),YES,,None,
type_of_water_source,varchar(64),YES,,None,
true_water_source_score,int,YES,,None,
statements,varchar(255),YES,,None,


In [13]:
%%sql

DESCRIBE water_quality

 * mysql+pymysql://root:***@localhost:3309/md_water_services
3 rows affected.


Field,Type,Null,Key,Default,Extra
record_id,int,NO,PRI,None,
subjective_quality_score,int,YES,,None,
visit_count,int,YES,,None,


In [14]:
%%sql

DESCRIBE water_source

 * mysql+pymysql://root:***@localhost:3309/md_water_services
3 rows affected.


Field,Type,Null,Key,Default,Extra
source_id,varchar(510),NO,PRI,None,
type_of_water_source,varchar(255),YES,,None,
number_of_people_served,int,YES,,None,


In [15]:
%%sql

DESCRIBE visits

 * mysql+pymysql://root:***@localhost:3309/md_water_services
7 rows affected.


Field,Type,Null,Key,Default,Extra
record_id,int,NO,PRI,None,
location_id,varchar(255),YES,MUL,None,
source_id,varchar(510),YES,MUL,None,
time_of_record,datetime,YES,,None,
visit_count,int,YES,,None,
time_in_queue,int,YES,,None,
assigned_employee_id,int,YES,MUL,None,


### JOIN - Linking Audit Data to Survey Records
To enable comparison, the auditor’s report is joined with the `visits` table using `location_id`. This allows each audited location to be mapped to its corresponding visit records.
The `record_id` obtained from this 'join' is then used to link to the water quality table for score comparison.

In [16]:
%%sql

SELECT  auditor.location_id,
        auditor.true_water_source_score, 
        vs.location_id as visit_location,
          vs.record_id  
FROM auditor_report as auditor
JOIN visits as vs ON vs.location_id = auditor.location_id
LIMIT 4;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
4 rows affected.


location_id,true_water_source_score,visit_location,record_id
SoRu34980,1,SoRu34980,5185
AkRu08112,3,AkRu08112,59367
AkLu02044,0,AkLu02044,37379
AkHa00421,3,AkHa00421,51627


With the `record_id` now available for each location, the next step is to retrieve the corresponding water quality scores from the `water_quality` table, with a specific focus on the `subjective_quality_score`.
This is achieved by joining the `visits` table with the `water_quality` table using `record_id` as the linking key, enabling direct comparison between survey and audit assessments.

In [17]:
%%sql

SELECT  auditor.location_id,
        wq.record_id,
        auditor.true_water_source_score as auditor_score,
        wq.subjective_quality_score as surveyor_score       
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
2698 rows affected.


location_id,record_id,auditor_score,surveyor_score
SoRu34980,5185,1,1
AkRu08112,59367,3,3
AkLu02044,37379,0,0
AkHa00421,51627,3,3
SoRu35221,28758,0,0
HaAm16170,31048,1,1
AkRu04812,1513,3,3
AkRu08304,1218,3,3
AkRu05107,8322,2,2
AkRu05215,21160,3,10


In [18]:
%%sql

SELECT  auditor.location_id,
        wq.record_id,
        auditor.true_water_source_score as auditor_score,
        wq.subjective_quality_score as surveyor_score       
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
WHERE auditor.true_water_source_score = wq.subjective_quality_score;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
2505 rows affected.


location_id,record_id,auditor_score,surveyor_score
SoRu34980,5185,1,1
AkRu08112,59367,3,3
AkLu02044,37379,0,0
AkHa00421,51627,3,3
SoRu35221,28758,0,0
HaAm16170,31048,1,1
AkRu04812,1513,3,3
AkRu08304,1218,3,3
AkRu05107,8322,2,2
HaDe16541,13070,2,2


The query returns 2,505 rows, which exceeds the expected number of unique locations. This discrepancy arises because some locations were visited multiple times, resulting in duplicate records after the joins. To ensure a one-to-one comparison, the dataset is filtered to include only initial visits by applying `visit_count = 1` in the `WHERE` clause. This removes duplicate entries and ensures that each location is represented once.

In [20]:
%%sql

SELECT  auditor.location_id,
        wq.record_id,
        auditor.true_water_source_score as auditor_score,
        wq.subjective_quality_score as surveyor_score       
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
WHERE auditor.true_water_source_score = wq.subjective_quality_score
AND vs.visit_count= 1;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
1518 rows affected.


location_id,record_id,auditor_score,surveyor_score
SoRu34980,5185,1,1
AkRu08112,59367,3,3
AkLu02044,37379,0,0
AkHa00421,51627,3,3
SoRu35221,28758,0,0
HaAm16170,31048,1,1
AkRu04812,1513,3,3
AkRu08304,1218,3,3
AkRu05107,8322,2,2
HaDe16541,13070,2,2


After removing duplicate records, the dataset is reduced to 1,518 valid entries. Given that the auditor reviewed 1,620 sites, this indicates a high level of agreement between the auditor’s assessments and the original survey data. Specifically, approximately 94% (1,518 out of 1,620) of the audited records are consistent, suggesting strong overall data reliability. However, this also implies that 102 records contain discrepancies.The next step is to isolate and analyse these inconsistent records to better understand the nature and potential causes of these differences.

In [21]:
%%sql

SELECT  auditor.location_id,
        wq.record_id,
        auditor.true_water_source_score as auditor_score,
        wq.subjective_quality_score as surveyor_score       
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
WHERE auditor.true_water_source_score != wq.subjective_quality_score
AND vs.visit_count= 1;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
102 rows affected.


location_id,record_id,auditor_score,surveyor_score
AkRu05215,21160,3,10
KiRu29290,7938,3,10
KiHa22748,43140,9,10
SoRu37841,18495,6,10
KiRu27884,33931,1,10
KiZu31170,17950,9,10
KiZu31370,36864,3,10
AkRu06495,45924,2,10
HaRu17528,30524,1,10
SoRu38331,13192,3,10


Given that this dataset was used in earlier analyses, it is important to verify that previous findings remain valid in light of the identified discrepancies. While water quality scores were not heavily relied upon, the analysis depended significantly on the `type_of_water_source`. Therefore, it is necessary to validate the consistency of this field across datasets.
To perform this check, the `type_of_water_source` column is retrieved from the `water_source` table and labelled as `survey_source`, using `source_id` as the join key. This is then compared with the corresponding `type_of_water_source` from the `auditor_report`, labelled as `auditor_source`, to identify any inconsistencies.

In [22]:
%%sql

SELECT  auditor.location_id,
        wq.record_id,
        ws.type_of_water_source as survey_source,
        auditor.type_of_water_source as auditor_source,
        auditor.true_water_source_score as auditor_score,
        wq.subjective_quality_score as surveyor_score       
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
JOIN water_source as ws ON ws.source_id =vs.source_id
WHERE auditor.true_water_source_score != wq.subjective_quality_score
AND vs.visit_count= 1;

 * mysql+pymysql://root:***@localhost:3309/md_water_services
102 rows affected.


location_id,record_id,survey_source,auditor_source,auditor_score,surveyor_score
AkRu05215,21160,well,well,3,10
KiRu29290,7938,shared_tap,shared_tap,3,10
KiHa22748,43140,tap_in_home_broken,tap_in_home_broken,9,10
SoRu37841,18495,shared_tap,shared_tap,6,10
KiRu27884,33931,well,well,1,10
KiZu31170,17950,tap_in_home_broken,tap_in_home_broken,9,10
KiZu31370,36864,shared_tap,shared_tap,3,10
AkRu06495,45924,well,well,2,10
HaRu17528,30524,well,well,1,10
SoRu38331,13192,shared_tap,shared_tap,3,10


### Linking Records to Employees

The next step is to investigate the potential sources of the identified discrepancies. The inconsistent records suggest that, in some cases, water quality scores were recorded inaccurately during data collection.

There are several possible explanations for these discrepancies. They may arise from normal human error, which is expected in field-based data collection. Alternatively, they may indicate systematic issues that require closer examination.

To better understand the origin of these inconsistencies, the analysis links each incorrect record to the corresponding employee responsible for data collection. This is achieved by joining the `assigned_employee_id` from the `visits` table to the current result set.

By doing so, it becomes possible to identify which employees are associated with the 102 inconsistent records and to analyse patterns in error distribution.

In [23]:
%%sql

SELECT auditor.true_water_source_score as auditors_score,
        wq.subjective_quality_score as surveyors_score,
        vs.record_id as visit_record_id,
        ee.assigned_employee_id,
        vs.location_id 
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
JOIN employee as ee ON ee.assigned_employee_id = vs.assigned_employee_id
WHERE auditor.true_water_source_score != wq.subjective_quality_score
AND vs.visit_count= 1


 * mysql+pymysql://root:***@localhost:3309/md_water_services
102 rows affected.


auditors_score,surveyors_score,visit_record_id,assigned_employee_id,location_id
3,10,21160,34,AkRu05215
3,10,7938,1,KiRu29290
9,10,43140,1,KiHa22748
6,10,18495,34,SoRu37841
1,10,33931,1,KiRu27884
9,10,17950,5,KiZu31170
3,10,36864,48,KiZu31370
2,10,45924,1,AkRu06495
1,10,30524,18,HaRu17528
3,10,13192,5,SoRu38331


The inconsistent records can now be linked to the employees responsible for capturing them. However, employee IDs alone are not sufficiently interpretable for analysis, so the corresponding employee names are retrieved from the `employee` table to improve clarity.

Given the complexity of the query at this stage, it is practical to store the result as a Common Table Expression (CTE). This improves readability and makes subsequent analysis more efficient by allowing the result set to be referenced as though it were a temporary table.

A suitable name for this CTE is `Incorrect_records`. Once defined, it should be verified by running:

`SELECT * FROM Incorrect_records`

to confirm that it returns the expected result set.

## CTE

In [24]:
%%sql

With Incorrect_records AS ( SELECT auditor.true_water_source_score as auditors_score,
        wq.subjective_quality_score as surveyors_score,
        vs.record_id as visit_record_id,
        ee.employee_name,
        vs.location_id 
FROM water_quality as wq
JOIN visits as vs ON vs.record_id = wq.record_id
JOIN auditor_report as auditor ON auditor.location_id = vs.location_id
JOIN employee as ee ON ee.assigned_employee_id = vs.assigned_employee_id
WHERE auditor.true_water_source_score != wq.subjective_quality_score
AND vs.visit_count= 1),

    error_count AS (SELECT 
      employee_name,
    Count(*) As number_of_mistakes
FROM Incorrect_records
GROUP BY employee_name
ORDER BY number_of_mistakes DESC),

    avg_error_count_per_empl AS (SELECT
    AVG(number_of_mistakes)
FROM error_count)

SELECT
employee_name,
number_of_mistakes
FROM
error_count
WHERE
number_of_mistakes > (SELECT
    AVG(number_of_mistakes)
FROM error_count);

 * mysql+pymysql://root:***@localhost:3309/md_water_services
4 rows affected.


employee_name,number_of_mistakes
Bello Azibo,26
Malachi Mavuso,21
Zuriel Matembo,17
Lalitha Kaburi,7


### Gathering Evidence

The distribution of discrepancies indicates that while most surveyors have relatively low error counts, a small number exhibit significantly higher rates of inconsistencies. This variation warrants further investigation.

At this stage, it is important to consider multiple explanations. Elevated error rates may be attributed to normal human error or differences in judgement during data collection. However, consistently high deviation from the average may also suggest underlying systematic issues that require closer examination.

To strengthen the analysis, both quantitative and qualitative evidence are incorporated. In addition to evaluating error frequency, the auditor’s recorded statements provide contextual insights from field observations. By combining these two sources, the analysis aims to develop a more comprehensive and reliable assessment of the observed discrepancies.

In [26]:
%%sql

CREATE VIEW Incorrect_records2 AS (
SELECT
auditor_report.location_id,
visits.record_id,
employee.employee_name,
auditor_report.true_water_source_score AS auditor_score,
wq.subjective_quality_score AS surveyor_score,
auditor_report.statements AS statements
FROM
auditor_report
JOIN
visits
ON auditor_report.location_id = visits.location_id
JOIN
water_quality AS wq
ON visits.record_id = wq.record_id
JOIN
employee
ON employee.assigned_employee_id = visits.assigned_employee_id
WHERE
visits.visit_count =1
AND auditor_report.true_water_source_score != wq.subjective_quality_score);


 * mysql+pymysql://root:***@localhost:3309/md_water_services
0 rows affected.


[]

In [27]:
%%sql

WITH error_count AS (
  SELECT
    employee_name,
    COUNT(*) AS number_of_mistakes
  FROM
    Incorrect_records2
  GROUP BY
    employee_name
),
suspect_list AS (
  SELECT
    employee_name,
    number_of_mistakes
  FROM
    error_count
  WHERE
    number_of_mistakes > (SELECT AVG(number_of_mistakes) FROM error_count)
)

    SELECT
  employee_name,
  location_id,
  statements
FROM
  Incorrect_records2
WHERE
  employee_name IN (SELECT employee_name FROM suspect_list);


 * mysql+pymysql://root:***@localhost:3309/md_water_services
71 rows affected.


employee_name,location_id,statements
Bello Azibo,KiRu29290,"A young artist sketches the faces in the queue, capturing the weariness of daily hours spent waiting for water."
Bello Azibo,KiHa22748,"A young girl's hopeful eyes are clouded by mistrust, her innocence tarnished by the corrupt system."
Bello Azibo,KiRu27884,"A traditional healer's empathy turns to bitterness, knowing that corrupt practices harm her community."
Zuriel Matembo,KiZu31170,"A community leader stood with his people, expressing concern for the water quality and the time lost in queues."","
Bello Azibo,AkRu06495,"A healthcare worker in the queue expressed fears about water-borne diseases, her face etched with worry."","
Zuriel Matembo,SoRu38331,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Malachi Mavuso,AmAm09607,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Zuriel Matembo,AkHa00314,"A street vendor's sales suffer from time spent waiting, her concern for the water's quality affecting her products."
Malachi Mavuso,KiRu26598,"A teenager's dreams are tempered by reality, her future threatened by the corrupt practices she sees around her."
Bello Azibo,KiIs23853,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The mention of cash changing hands further tainted their perception.


In [28]:
%%sql

SELECT
  employee_name,
  location_id,
  statements
FROM
  Incorrect_records2
WHERE
  statements LIKE '%cash%'


 * mysql+pymysql://root:***@localhost:3309/md_water_services
19 rows affected.


employee_name,location_id,statements
Zuriel Matembo,SoRu38331,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Malachi Mavuso,AmAm09607,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Bello Azibo,KiIs23853,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The mention of cash changing hands further tainted their perception.
Bello Azibo,HaSe21323,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Zuriel Matembo,AkRu05880,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The allusion to cash changing hands deepened their skepticism.
Bello Azibo,KiRu27065,Villagers expressed their discomfort with an official who displayed a haughty demeanor and negligence. The mention of cash transactions deepened their growing sense of unease.
Malachi Mavuso,KiRu25347,Villagers expressed their discontent with an official who appeared dismissive and neglectful. The mention of cash changing hands added to their growing sense of distrust.
Zuriel Matembo,SoIl32575,Villagers recounted unsettling encounters with an official known for their arrogance and avoidance of responsibilities. The mention of cash changing hands added to their apprehension and distrust.
Bello Azibo,AkRu04508,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Lalitha Kaburi,AkRu07310,"Villagers spoke of their unsettling encounters with an official who seemed indifferent and uninterested, hinting at potential improprieties involving cash exchanges."


In [29]:
%%sql

WITH error_count AS (
  SELECT
    employee_name,
    COUNT(*) AS number_of_mistakes
  FROM
    Incorrect_records2
  GROUP BY
    employee_name
),
suspect_list AS (
    SELECT ec1.employee_name, ec1.number_of_mistakes
    FROM error_count ec1
    WHERE ec1.number_of_mistakes >= (
        SELECT AVG(ec2.number_of_mistakes)
        FROM error_count ec2
        WHERE ec2.employee_name = ec1.employee_name))

   
SELECT
  employee_name,
  location_id,
  statements
FROM
  Incorrect_records2
WHERE
  statements LIKE '%cash%'
 AND employee_name  IN (SELECT employee_name FROM suspect_list)
 


 * mysql+pymysql://root:***@localhost:3309/md_water_services
19 rows affected.


employee_name,location_id,statements
Zuriel Matembo,SoRu38331,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Malachi Mavuso,AmAm09607,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Bello Azibo,KiIs23853,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The mention of cash changing hands further tainted their perception.
Bello Azibo,HaSe21323,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Zuriel Matembo,AkRu05880,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The allusion to cash changing hands deepened their skepticism.
Bello Azibo,KiRu27065,Villagers expressed their discomfort with an official who displayed a haughty demeanor and negligence. The mention of cash transactions deepened their growing sense of unease.
Malachi Mavuso,KiRu25347,Villagers expressed their discontent with an official who appeared dismissive and neglectful. The mention of cash changing hands added to their growing sense of distrust.
Zuriel Matembo,SoIl32575,Villagers recounted unsettling encounters with an official known for their arrogance and avoidance of responsibilities. The mention of cash changing hands added to their apprehension and distrust.
Bello Azibo,AkRu04508,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Lalitha Kaburi,AkRu07310,"Villagers spoke of their unsettling encounters with an official who seemed indifferent and uninterested, hinting at potential improprieties involving cash exchanges."


### Conclusion

The analysis identifies four employees Zuriel Matembo, Malachi Mavuso, Bello Azibo, and Lalitha Kaburi whose records exhibit patterns that warrant further review.
Two key observations support this finding:
1. Each of these employees has an above-average number of inconsistencies in recorded data.  
2. Their associated records include qualitative statements that raise concerns, with no similar patterns observed among other employees.

While these findings do not constitute definitive evidence of misconduct, the combination of elevated error rates and supporting contextual indicators is sufficiently concerning to justify further investigation.
These results should be formally flagged for review to ensure data integrity and accountability within the system.